tests/core/test_library.py

```diff
@@ -33,6 +37,50 @@ def setUp(self):  # noqa: N802
         self.core = core.Core(mixer=None, backends=[
             self.backend1, self.backend2, self.backend3])
 
+    def test_get_images_returns_empty_dict_for_no_uris(self):
+        self.assertEqual({}, self.core.library.get_images([]))
+
+    def test_get_images_returns_empty_result_for_unknown_uri(self):
+        result = self.core.library.get_images(['dummy4:track'])
+        self.assertEqual({'dummy4:track': tuple()}, result)
+
+    def test_get_images_returns_empty_result_for_library_less_uri(self):
+        result = self.core.library.get_images(['dummy3:track'])
+        self.assertEqual({'dummy3:track': tuple()}, result)
+
+    def test_get_images_maps_uri_to_backend(self):
+        self.core.library.get_images(['dummy1:track'])
+        self.library1.get_images.assert_called_once_with(['dummy1:track'])
+        self.library2.get_images.assert_not_called()
+
+    def test_get_images_maps_uri_to_backends(self):
+        self.core.library.get_images(['dummy1:track', 'dummy2:track'])
+        self.library1.get_images.assert_called_once_with(['dummy1:track'])
+        self.library2.get_images.assert_called_once_with(['dummy2:track'])
+
+    def test_get_images_returns_images(self):
+        self.library1.get_images().get.return_value = {
+            'dummy1:track': [Image(uri='uri')]}
+        self.library1.get_images.reset_mock()
+
+        result = self.core.library.get_images(['dummy1:track'])
+        self.assertEqual({'dummy1:track': (Image(uri='uri'),)}, result)
+
+    def test_get_images_merges_results(self):
+        self.library1.get_images().get.return_value = {
+            'dummy1:track': [Image(uri='uri1')]}
+        self.library1.get_images.reset_mock()
+        self.library2.get_images().get.return_value = {
```

Comment:
I know we've used this pattern some places in our tests, but the proper way to do this (without registering an extra call to `get_images()`) is to `s/get_images().get/get_images.return_value.get/`.